File: FTP Client for WinForms using Ultimate FTP library/[C#]-FTP Client for WinForms using Ultimate FTP library/C#/ClientController_Fields.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace ClientSample
{
    partial class ClientController
    {
        private bool _reconnecting; // Indicates whether the connection is closed unexpectedly and the app is about to reconnect.

        private ConnectionState _state; // The connection state.
        private ServerProtocol _serverType;

        private string _currentDirectory;
        string _currentLocalDirectory;
        private bool _aborting;

        public string CurrentLocalDirectory
        {
            get { return _currentLocalDirectory; }
        }

        public string CurrentRemoteDirectory
        {
            get { return _currentDirectory; }
        }
    }
}

File: FTP Client for WinForms using Ultimate FTP library/[VB]-FTP Client for WinForms using Ultimate FTP library/VB/ClientController_Fields.vb
Imports System.Text
Imports System.Threading

Namespace ClientSample
	Partial Friend Class ClientController
		Private _reconnecting As Boolean ' Indicates whether the connection is closed unexpectedly and the app is about to reconnect.

		Private _state As ConnectionState ' The connection state.
		Private _serverType As ServerProtocol

		Private _currentDirectory As String
		Private _currentLocalDirectory As String
		Private _aborting As Boolean

		Public ReadOnly Property CurrentLocalDirectory() As String
			Get
				Return _currentLocalDirectory
			End Get
		End Property

		Public ReadOnly Property CurrentRemoteDirectory() As String
			Get
				Return _currentDirectory
			End Get
		End Property
	End Class
End Namespace
